source/particles/particle_handler.cc

```diff
@@ -1067,6 +1067,11 @@ namespace Particles
                                                           cell_centers);
                     });
 
+          // make a copy of the current cell, since we will modify the variable
+          // in the following current_cell but we need a backup in the case
+          // the particle is not found
+          const auto current_cell_old = current_cell;
```

Comment:
Shouldn't we call this `previous_cell_of_particle`?

Reply:
Done.